pkgs/games/starsector/default.nix

```
@@ -62,12 +62,16 @@ stdenv.mkDerivation rec {
   # it tries to run everything with relative paths, which makes it CWD dependent
   # also point mod, screenshot, and save directory to $XDG_DATA_HOME
   # additionally, add some GC options to improve performance of the game
+  # lastly, pass-through CLI args ($@) to the underlying java call, so the user
+  # can override JVM arguments (e.g. setting a higher -Xmx value for modded
+  # usage)
   postPatch = ''
     substituteInPlace starsector.sh \
       --replace "./jre_linux/bin/java" "${openjdk}/bin/java" \
       --replace "./native/linux" "$out/share/starsector/native/linux" \
       --replace "=." "=\''${XDG_DATA_HOME:-\$HOME/.local/share}/starsector" \
-      --replace "-XX:+CompilerThreadHintNoPreempt" "-XX:+UnlockDiagnosticVMOptions -XX:-BytecodeVerificationRemote -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+CMSConcurrentMTEnabled -XX:+DisableExplicitGC"
+      --replace "-XX:+CompilerThreadHintNoPreempt" "-XX:+UnlockDiagnosticVMOptions -XX:-BytecodeVerificationRemote -XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+CMSConcurrentMTEnabled -XX:+DisableExplicitGC" \
+      --replace "com.fs.starfarer.StarfarerLauncher" "com.fs.starfarer.StarfarerLauncher \"\$@\""
```

By inserting the cli arguments `$@` after the main class name `com.fs.starfarer.StarfarerLauncher` you are passing the arguments to the program being ran (starsector) and not the JVM.

This behaviour can be confirmed by running `java --help` and reading the relevant parts:
```
Usage: java [options] <mainclass> [args...]
...
 Arguments following the main class, source file, -jar <jarfile>,
 -m or --module <module>/<mainclass> are passed as the arguments to
 main class.
```

The simple solution would be to move the arguments to before the main class name, as follows:

```suggestion
      --replace "com.fs.starfarer.StarfarerLauncher" "\"\$@\" com.fs.starfarer.StarfarerLauncher"
```

---

@rafaelrc7 another thing is that without the `-XX:+UnlockDiagnosticVMOptions -XX:-BytecodeVerificationRemote` the game would not run (unless they fixed their obfuscator, idk), I see that you made it into `--replace-warn`, I feel like it should be added as part of some `--replace-fail`

Or just give up and write our own startup script from scratch instead of replacing the original by bits and pieces so much :)

---

@necauqua I agree, I noticed this issue after your comments on the previous PR. I was planning in fixing it soon. If @Shados does not respond soon, I plan on opening a new PR with fixes for both the CLI args passthrough and the oversight you mentioned.

---

@rafaelrc7 it'd be for the best if you go ahead and do that, I'm unfortunately pretty busy at the moment. I'll close this off.

---

Ok, thanks for the reply!